chain/pool/txpool.go

```diff
@@ -678,3 +707,109 @@ func (tp *TxnPool) GetNonceByTxnPool(addr common.Uint160) (uint64, error) {
 func shortHashToKey(shortHash []byte) string {
 	return string(shortHash)
 }
+
+// sync txn
+// update twtFingerprint
+func (tp *TxnPool) updateTwtFingerprint(txnHash common.Uint256) {
+	byteTxnHash := txnHash.ToArray()
+	if len(tp.twtFingerprint) == 0 {
+		tp.twtFingerprint = byteTxnHash
+	} else {
+		for i := 0; i < len(byteTxnHash); i++ {
+			tp.twtFingerprint[i] ^= byteTxnHash[i]
+		}
+	}
+}
+
+// when append new txn to pool, we add this txn pointer to txtWithTime map too.
+func (tp *TxnPool) AppendTwt(txn *transaction.Transaction) {
+
+	now := time.Now().UnixMilli()
+
+	twt := TxnWithTime{
+		ArriveTime: now,
+		Txn:        txn,
+	}
+
+	tp.twtMapMu.Lock()
+	tp.twtMap.Set(txn.Hash(), twt)
+	tp.updateTwtFingerprint(txn.Hash())
+	tp.twtMapMu.Unlock()
+
+	atomic.AddInt32(&tp.twtCount, 1)
```

Comment:
Don't use `atomic` at all, protect `twtCount` with the lock

Reply:
yes. fixed.